typings/join-with-dash.ts

```diff
@@ -0,0 +1,10 @@
+export type JoinWithDash<T extends string[]> = T extends [
+  infer First extends string,
+  ...infer Rest extends string[],
+]
+  ? Rest extends []
+    ? `${First}`
+    : `${WithDashSuffixOrEmpty<First>}${JoinWithDash<Rest>}`
+  : never
+
+type WithDashSuffixOrEmpty<T extends string> = `${T}-` | ''
```

Comment:
Looks good. 

I have an idea. Maybe it makes sense to add a second `Separator` parameter?


```ts
export type Join<
  T extends string[],
  Separator extends string = '',
> = T extends [infer First extends string, ...infer Rest extends string[]]
  ? `${First}${Rest['length'] extends 0 ? '' : `${Separator}${Join<Rest, Separator>}`}`
  : ''

export type JoinWithDash<T extends string[]> = Join<T, '-'>
```

Reply:
@azat-io 👍 Improved in a25540611d7d34ca64ff06c6bfbc472b201a33c2.